pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -224,9 +224,23 @@ private CompletableFuture<Reader<T>> newReader(String partition) {
                 .readCompacted(true)
                 .poolMessages(true)
                 .createAsync()
+                .thenCompose(this::cacheNewReader)
                 .thenCompose(this::readAllExistingMessages);
     }
 
+    private CompletableFuture<Reader<T>> cacheNewReader(Reader<T> reader) {
+        CompletableFuture<Reader<T>> future = new CompletableFuture<>();
+        if (this.readers.containsKey(reader.getTopic())) {
```

Comment:
do we need to close the reader if the readers already cached one?

Reply:
Here, How about checking cache first before creating a new reader ?

Reply:
Here, How about check  cache first before creating reader?